pkg/utils/resources/suite_test.go

```diff
@@ -0,0 +1,409 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestResources(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Resources Suite")
+}
+
+var _ = Describe("Resources Handling", func() {
+	Context("Resource Calculations", func() {
+		It("should schedule based on the sum of containers and sidecarContainers", func() {
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1"), v1.ResourceMemory: resource.MustParse("2Gi")},
+						},
+					},
+				},
+			})
+			podResources := Ceiling(pod)
+			Expect(podResources.Requests.Cpu().Value()).To(BeNumerically("==", 3))
+			Expect(podResources.Requests.Memory().Value()).To(BeNumerically("==", 3221225472))
+			Expect(podResources.Limits.Cpu().Value()).To(BeNumerically("==", 6))
+			Expect(podResources.Limits.Memory().Value()).To(BeNumerically("==", 3221225472))
+		})
+
+		It("should schedule based on the sum of containers , sidecarContainers and overhead with max of initcontainers", func() {
```

Comment:
Reading through the test: Does this test case describe what the test is doing well? In this test case, we aren't considering the "max of init containers", we are actually considered the native sidecar container. A test that ensured what the test description states would be one where it's just init containers and both init containers request more cpu/memory than the pod resource requests plus overhead